docs/qwik-testing-library/example.mdx

```diff
@@ -0,0 +1,101 @@
+title: Example
+sidebar_label: Example
+---
+
+Below are some examples of how to use the Qwik Testing Library to tests your
+Qwik components.
+
+You can also learn more about the [**queries**][tl-queries-docs] and [**user
+events**][tl-user-events-docs] to help you write your tests.
+
+[tl-queries-docs]: ../queries/about.mdx
+[tl-user-events-docs]: ../user-event/intro.mdx
+
+## Qwikstart
+
+This is a minimal setup to get you started, with line-by-line explanations.
+
+```tsx title="counter.spec.tsx"
+// import qwik-testing methods
+import {screen, render, waitFor} from '@noma.to/qwik-testing-library'
+// import the userEvent methods to interact with the DOM
+import {userEvent} from '@testing-library/user-event'
+// import the component to be tested
+import {Counter} from './counter'
+
+// describe the test suite
+describe('<Counter />', () => {
+  // describe the test case
+  it('should increment the counter', async () => {
+    // render the component into the DOM
+    await render(<Counter />)
+
+    // retrieve the 'increment count' button
+    const incrementBtn = screen.getByRole('button', {name: /increment count/})
+    // click the button twice
+    await userEvent.click(incrementBtn)
+    await userEvent.click(incrementBtn)
+
+    // assert that the counter is now 2
+    await waitFor(() => expect(screen.getByText(/count:2/)).toBeInTheDocument())
+  })
+})
+```
+
+## Qwik City - `server$` calls
+
+If one of your Qwik components uses `server$` calls, your tests might fail with
+a rather cryptic message (e.g.
+`QWIK ERROR __vite_ssr_import_0__.myServerFunctionQrl is not a function` or
+`QWIK ERROR Failed to parse URL from ?qfunc=DNpotUma33o`).
+
+We're happy to discuss it on [Discord][discord], but we consider this failure to
+be a good thing: your components should be tested in isolation, so you will be
+forced to mock your server functions.
+
+Here is an example of how to test a component that uses `server$` calls:
+
+```ts title="~/server/blog-post.ts"
+import {server$} from '@builder.io/qwik-city'
+import {BlogPost} from '~/lib/blog-post'
+
+export const getLatestPosts$ = server$(function (): Promise<BlogPost> {
+  // get the latest posts
+  return Promise.resolve([])
+})
+```
+
+```tsx title="~/components/latest-post-list.tsx"
+import {render, screen, waitFor} from '@noma.to/qwik-testing-library'
+import {LatestPostList} from './latest-post-list'
+
+vi.mock('~/server/blog-posts', () => ({
+  // the mocked function should end with `Qrl` instead of `$`
+  getLatestPostsQrl: () => {
+    return Promise.resolve([
+      {id: 'post-1', title: 'Post 1'},
+      {id: 'post-2', title: 'Post 2'},
+    ])
+  },
+}))
+
+describe('<LatestPostList />', () => {
+  it('should render the latest posts', async () => {
+    await render(<LatestPostList />)
+
+    await waitFor(() =>
+      expect(screen.queryAllByRole('listitem')).toHaveLength(2),
+    )
+  })
+})
+```
+
+Notice how the mocked function is ending with `Qrl` instead of `$`, despite
```

Comment:
This seems to me like an implementation detail that we should maybe think about hiding from the user? Also, is this aligned with the qwik team? If this will change the entire mocking solution will break.

Reply:
I agree that it is inconvenient for the end user to have to think about this when mocking those kind of function.

For the context: this is how Qwik works and qwik devs are [aware of this](https://qwik.dev/docs/advanced/dollar/). In Qwik, this is not really considered as an implementation detail as devs can consciously choose one approach (`Qrl`) or the other `$`. It's just that the `$` is way more convenient and less boilerplate.

I'll try to think of a way to apply the same magic (`$` -> `Qrl`) when mocking entire modules. I'll keep you updated.

Reply:
Thanks.. This of course is not a blocker, just a comment :)